generator/mavgen.py

```diff
@@ -109,10 +109,7 @@ def mavgen_validate(xmlfile):
             x.largest_payload = max(x.largest_payload, xml[-1].largest_payload)
 
     # work out max payload size across all includes
-    largest_payload = 0
-    for x in xml:
-        if x.largest_payload > largest_payload:
-            largest_payload = x.largest_payload
+    largest_payload = max(x.largest_payload for x in xml)
```

Comment:
Nice change!  However, it seems the xml list might be empty, and that isn't going to end well.  We need some protection in for this case.